src/Platform/Microsoft.Testing.Platform/ServerMode/JsonRpc/ServerModePerCallOutputDevice.cs

```diff
@@ -19,11 +34,72 @@ internal class ServerModePerCallOutputDevice : IPlatformOutputDevice
 
     public Task DisplayAfterSessionEndRunAsync() => Task.CompletedTask;
 
-    public Task DisplayAsync(IOutputDeviceDataProducer producer, IOutputDeviceData data) => Task.CompletedTask;
+    public async Task DisplayAsync(IOutputDeviceDataProducer producer, IOutputDeviceData data)
+    {
+        using (await _asyncMonitor.LockAsync(TimeoutHelper.DefaultHangTimeSpanTimeout))
```

Comment:
I don't think we need this lock, we could just forward down and the thing below is responsible for that.

@MarcoRossignoli shall we lock to ensure ordering below?